x-pack/plugins/security_solution/public/common/components/alerts_treemap/lib/layers/index.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Datum } from '@elastic/charts';
+
+import { getFillColor } from '../chart_palette';
+import { getLabel } from '../labels';
+
+export interface DataName {
+  dataName: string;
+}
+
+export interface Path {
+  index: number;
+  value: string;
+}
+
+export interface FillColorDatum {
+  path?: Path[];
+}
+
+// common functions used by getLayersOneDimension and getLayersMultiDimensional:
+const valueFormatter = (d: number) => `${d}`;
+const groupByRollup = (d: Datum) => d.key;
+
+/**
+ * Extracts the first group name from the data representing the second group
+ */
+export const getGroupFromPath = (datum: FillColorDatum): string | undefined => {
+  const OFFSET_FROM_END = 2; // The offset from the end of the path array containing the group
```

Comment:
Random question, why is it the second to last item in the array?

Reply:
> Random question, why is it the second to last item in the array?

The values of a `datum`, (the argument to this function), and specifically its `path` property, are defined by `Elastic Charts`.

The total number of items in the "front" of the `path` is variable, but the group is always available offset from the end of the  `path` array, per the example below:

```json
[
    {
        "index": 0,
        "value": "__null_small_multiples_key__"
    },
    {
        "index": 0,
        "value": "__root_key__"
    },
    {
        "index": 4,
        "value": "mimikatz process started"
    },
    {
        "index": 2,
        "value": "Host-0ucgxjd9lk"
    }
]
```